test_crates/inherent_associated_pub_const_missing/old/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+// Test Cases where #[doc(hidden)] is neither added or removed
+pub struct StructA {}
+
+impl StructA {
+    // Basic Test case should be caught
+    pub const PublicConstantA: i32 = 0;
+    pub const PublicConstantB: i32 = 0;
+    // Const that was #[doc(hidden)] will be removed
+    #[doc(hidden)]
+    pub const DocHiddenConstantA: i32 = 0;
+    // Should Be caught on renaming
+    pub const PublicConstantRenameA: i32 = 0;
+    // Should not be caught on removing since its not pub
+    const ConstantA: i32 = 0;
+}
+
+struct StructB {}
+
+impl StructB {
+    // Should not be caught since StructB is not pub
+    pub const PublicConstantC: i32 = 0;
+    pub const PublicConstantRenameB: i32 = 0;
+    const ConstantB: i32 = 0;
+}
+
+#[doc(hidden)]
+pub struct StructC {}
+
+impl StructC {
+    // Should not be caught on removing or renaming since the struct #[doc(hidden)]
+    pub const PublicConstantD: i32 = 0;
+    pub const PublicConstantRenameC: i32 = 0;
+}
+
+pub struct StructD {}
+
+#[doc(hidden)]
+impl StructD {
+    pub const PublicConstantE: i32 = 0;
+    pub const PublicConstantF: i32 = 0;
+}
```

Comment:
```suggestion
// The constants in this block shouldn't be flagged as removed,
// since they are `#[doc(hidden)]` as a result of the block itself.
#[doc(hidden)]
impl StructD {
    pub const PublicConstantE: i32 = 0;
    pub const PublicConstantF: i32 = 0;
}
```